File: 19주차/DE/24. Swap Nodes in Pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 function swapPairs(head) {
    let node = head;
    while(node !== null){
       if(node.next == null) break;
       let temp = node.next.val
       node.next.val = node.val;
       node.val = temp;
        node = node.next.next;
   }
    return head;
};

File: 19주차/KM/24-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode p=head;
        if(p == null || p.next == null)
            return head;
        ListNode prev = head;
        ListNode cur = head.next;
        ListNode next = cur.next;

        prev.next=swapPairs(next);
        cur.next=prev;
      
     
        return cur; 
    }
}